mantidproject/mantid
# Add Cropping Options to Eng Diff Calib Tab

**Description of work.**
Cropping can now be done when creating a new calibration.
The GUI now allows for cropping by bank (1 or 2), or by specific spectrum numbers. 

**To test:**
1. Open Engineering Diffraction 2 GUI
2. Make sure that cropping options are greyed out when Load is selected.
3. Select Create New again (Valid Options: V: 307521 CeO2: 305738)
4. Check that the combo box selector appears when the cropping check box is selected 
5. Check selecting custom spectra reveals a Text entry
6. Check that only sane values can be entered and error messages make sense. (1200-1400 should guarantee an output)
7. Check that outputs are restricted by the cropping and have sane names. 


Fixes #26849 

*This does not require release notes* because **the GUI is not user ready.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
